tests_external/selenium.mjs

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview smoke test for Selenium integration.
+ *
+ * Runs Selenium with the latest CfT + ChromeDriver + current Mapper version.
+ * Inspired by https://github.com/SeleniumHQ/selenium/blob/0c86525184355bddc44b6193ae7236f11a7fb129/javascript/node/selenium-webdriver/test/bidi/bidi_test.js#L300
+ */
+
+import {execSync} from 'child_process';
+import * as assert from 'node:assert';
+import {join} from 'path';
+
+import {Builder, ScriptManager, BrowsingContext} from 'selenium-webdriver';
+import chrome from 'selenium-webdriver/chrome.js';
+
+function installAndGetChromePath() {
+  let BROWSER_BIN = process.env.BROWSER_BIN;
+  if (!BROWSER_BIN) {
+    BROWSER_BIN = execSync(
+      ['node', join('tools', 'install-browser.mjs')].join(' ')
+    )
+      .toString()
+      .trim();
+  }
+  return BROWSER_BIN;
+}
+
+function installAndGetChromeDriverPath() {
+  let CHROME_DRIVER_BIN = process.env.CHROME_DRIVER_BIN;
+  if (!CHROME_DRIVER_BIN) {
+    CHROME_DRIVER_BIN = execSync(
+      ['node', join('tools', 'install-browser.mjs'), '--chrome-driver'].join(
+        ' '
+      )
+    )
+      .toString()
+      .trim();
+  }
+  return CHROME_DRIVER_BIN;
+}
+
+const chromePath = installAndGetChromePath();
+const chromeDriverPath = installAndGetChromeDriverPath();
+
+const chromeService = new chrome.ServiceBuilder(chromeDriverPath).addArguments(
+  '--bidi-mapper-path=lib/iife/mapperTab.js'
```

Comment:
This path should also be composed with `join`

Reply:
done